data/sensors_recent.php

```
@@ -193,59 +182,64 @@ function output_cell($rowspan, $data) {
-						$meta['gtw_eui'] = "";
-						if (isset($meta['gateway_ids']['eui']))
-							$meta['gtw_eui'] = strtolower($meta['gateway_ids']['eui']);
-						if ($meta['gtw_id'] == 'packetbroker')
-							$meta['gtw_id'] = 'unknown';
-						if (! isset($meta['snr'])) 
-							$meta['snr'] = $null_snr;
-
-						$gateways[] = $meta;
-					}
 	
-					$metadata = [
-						'gateways' => $gateways,
-						'frequency' => $settings['frequency'] / 1000000.,
-						'spreading_factor' => $settings['data_rate']['lora']['spreading_factor'],
-						'data_rate' => $settings['data_rate']['lora']['bandwidth'] / 1000,
-						'coding_rate' => $settings['coding_rate'],
-					];
+				$gateways = [];
+
+				// Convert TTN v3 to v2 format
+				foreach ($metadata as $meta) {
+					$meta['gtw_id'] = $meta['gateway_ids']['gateway_id'];
+					$meta['gtw_eui'] = "";
+					if (isset($meta['gateway_ids']['eui']))
+						$meta['gtw_eui'] = $meta['gateway_ids']['eui'];
+					if ($meta['gtw_id'] == 'packetbroker')
+						$meta['gtw_id'] = 'unknown';
+					if (!isset($meta['snr'])) 
+						$meta['snr'] = $null_snr;
+
+					$gateways[] = $meta;
 				}
-			}	// if ttn.v2 else ttn.v3
+	
+				$spreading_factor = $message['uplink_message']['settings']['data_rate']['lora']['spreading_factor'];
+				$bandwidth =        $message['uplink_message']['settings']['data_rate']['lora']['bandwidth'] / 1000;
+				$data_rate = sprintf("SF%sBW%s", $spreading_factor, $bandwidth);
+				$metadata = [
+					'gateways'    => $gateways,
+					'frequency'   => $message['uplink_message']['settings']['frequency'] / 1000000.,
+					'coding_rate' => $message['uplink_message']['settings']['coding_rate'],
+					'data_rate'   => $data_rate,
+					'counter'     => $message['uplink_message']['f_cnt'],
+				];
+			}	// if ttn.v1 else ttn.v2 else ttn.v3
 
 			$gateways = $metadata['gateways'];
+			if (count($gateways) > 3) $gateways = array($gateways[0], $gateways[1], $gateways[2]);	// now only showing 3 best-receiving gateways
```

I don't think this is a good idea, since this silently hides information. If we want to make the interface more compact (which would make sense, given there are so much gateways nowadays), I would rather suggest doing this on the browser by collapsing the extra gateways and allowing the user to expand the list. But I think this is out of scope for this PR, so please remove this again and then we can figure this out in a separate PR later.

---

I suggest a URL parameter like: ...&maxgw=3